test/linalg/bunchkaufman.jl

```diff
@@ -101,29 +101,25 @@ bimg  = randn(n,2)/2
                 end
             end
         end
-    end
-end
-
-
-@testset "Bunch-Kaufman factors of a singular matrix" begin
-    let As1 = ones(n, n)
-        As2 = complex(ones(n, n))
-        As3 = complex(ones(n, n))
-        As3[end, 1] += im
-        As3[1, end] -= im
-
-        for As = (As1, As2, As3)
-            for As in (As, view(As, 1:n, 1:n))
-                @testset for rook in (false, true)
-                    @testset for uplo in (:L, :U)
-                        F = bkfact(issymmetric(As) ? Symmetric(As, uplo) : Hermitian(As, uplo), rook)
-                        @test !LinAlg.issuccess(F)
-                        # test printing of this as well!
-                        bks = sprint(show, "text/plain", F)
-                        @test bks == "Failed factorization of type $(typeof(F))"
-                        @test det(F) == 0
-                        @test_throws LinAlg.SingularException inv(F)
-                        @test_throws LinAlg.SingularException F \ ones(size(As, 1))
+        if eltya <: BlasReal
+            As1 = ones(eltya, n, n)
+            As2 = complex(ones(eltya, n, n))
+            As3 = complex(ones(eltya, n, n))
+            As3[end, 1] += convert(complex(eltya), im)
```

Comment:
1. Could you try to add and subtract `im/2` instead of `im` here? I think that might about the 2x2 pivot and I'm suspecting that to cause the rounding error.
2. The `convert` here shouldn't be needed since you modify in place